ern-api-impl-gen/src/ApiImpl.js

```diff
@@ -0,0 +1,144 @@
+
+    // Creates a working directory to collect all the necessary files/directories for the api-impl generation.
+    createWorkingDirectory(paths.workingDirectory)
+    createPluginsDownloadDirectory(paths.pluginsDownloadDirectory)
+
+    await new ApiImplGen().generateApiImplementation(apiDependency, paths, reactNativeVersion, platforms)
+  } catch (e) {
+    log.debug('command failed performing cleanup.')
+    throw new Error(e)
+  }
+}
+
+async function createOutputDirectory (outputDirectoryPath: string, forceGenerate) {
+  if (!forceGenerate && fs.existsSync(outputDirectoryPath)) {
+    const {shouldRegenerate} = await inquirer.prompt(
+      {
+        type: `confirm`,
+        name: `shouldRegenerate`,
+        message: `An implementation directory already exists in ${outputDirectoryPath}. Do you want to delete this and regenerate this project?`,
+        default: false
+      }
+    )
+
+    if (!shouldRegenerate) {
+      throw Error(`An implementation directory already exists`)
+    } else {
+      forceGenerate = true
+    }
+  }
+
+  if (forceGenerate && fs.existsSync(outputDirectoryPath)) {
+    log.info(`Deleting the existing directory and recreating a new one in ${outputDirectoryPath}`)
+    shell.rm('-R', outputDirectoryPath)
+  } else {
+    log.debug(`creating output dir: ${outputDirectoryPath}`)
+  }
+  shell.mkdir('-p', outputDirectoryPath)
+}
+
+async function createNodePackage (
+  outputDirectoryPath: string,
+  apiDependency: Dependency,
+  nativeOnly: boolean,
+  hasConfig: boolean) {
+  let currentDirectory = process.cwd()
+  shell.cd(outputDirectoryPath)
+  await yarn.init()
+  await yarn.add(apiDependency.path)
+  shell.rm('-rf', path.join(outputDirectoryPath, 'node_modules'))
+  ernifyPackageJson(outputDirectoryPath, nativeOnly, hasConfig)
+  shell.cd(currentDirectory)
+}
+
+function createWorkingDirectory (workingDirectoryPath: string) {
+  shell.rm('-rf', workingDirectoryPath)
+  shell.mkdir('-p', workingDirectoryPath)
+  log.debug(`Working directory created: ${workingDirectoryPath}`)
+}
+
+function createPluginsDownloadDirectory (pluginsDownloadPath: string) {
+  shell.rm('-rf', pluginsDownloadPath)
+  shell.mkdir('-p', pluginsDownloadPath)
+}
+
+function formOutputDirectoryName (apiDependency: Dependency, outputDirectoryPath: string) {
+  let apiName = API_NAME_RE.exec(apiDependency.name)[1]
+  return outputDirectoryPath
+    ? path.join(outputDirectoryPath, `${apiName}-impl`)
+    : path.join(process.cwd(), `${apiName}-impl`)
+}
+
+function getPlatforms (nativeOnly: boolean): Array<string> {
+  return nativeOnly
+    ? [`android`, `ios`]
+    : [`js`]
+}
+
+function ernifyPackageJson (outputDirectoryPath, nativeOnly, hasConfig) {
+  const packageJsonPath = path.join(outputDirectoryPath, 'package.json')
+  const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'))
+  const moduleType = nativeOnly ? `${ModuleTypes.NATIVE_API_IMPL}` : `${ModuleTypes.JS_API_IMPL}`
```

Comment:
no need to use template litteral here. 
`ModuleTypes.NATIVE_API_IMPL : ModuleTypes.JS_API_IMPL`
instead of
``` `${ModuleTypes.NATIVE_API_IMPL}` : `${ModuleTypes.JS_API_IMPL}` ```